netutils/platform_mapper.py

```diff
@@ -0,0 +1,231 @@
+    """
+    manufacturer = vendor.lower()
+    platform = platform.lower()
+    platform_obj = None
+
+    if platform in custom_nist_platforms.get(manufacturer, {}):
+        platform_obj = eval(  # pylint: disable=W0123 # nosec B307
+            f"{manufacturer.capitalize()}Platform(platform, version)"
+        )
+    else:
+        platform_obj = OSPlatform(manufacturer, platform, version)
+
+    return platform_obj
+
+
+class OSPlatform:
+    """Class to create a generic platform for identification and manipulation."""
+
+    def __init__(self, vendor: str, platform: str, version: str):
+        """Initialize OSPlatform Object with vendor, platform, and version."""
+        self.vendor = vendor
+        self.platform = platform
+        self.version = version
+
+    def get_info(self) -> t.Dict[str, str]:
+        r"""Display Platform Information.
+
+        Can be used with any Child class that inits super().
+
+        Returns:
+            Dictionary of platform breakdown
+
+        Examples:
+            >>> OSPlatform('cisco','nxos','15.1(7)').get_info()
+            {'vendor': 'cisco', 'platform': 'nxos', 'version': '15.1(7)'}
+
+            >>> JuniperPlatform('junos','12.1R3-S4.1').get_info()
+            {'vendor': 'juniper', 'platform': 'junos', 'version': '12.1R3-S4.1', 'isservice': True, 'ismaintenance': True, 'isfrs': False, 'isspecial': False, 'main': '12', 'minor': '1', 'type': 'R', 'build': '3', 'service': 'S', 'service_build': '4', 'service_respin': '1'}
+        """
+        return self.__dict__
+
+    def get_nist_urls(self, api_key: str) -> t.List[str]:
+        r"""Create a list of possible NIST Url strings.
+
+        Child models with NIST URL customizations need their own "get_nist_urls" method.
+
+        Args:
+            api_key: NIST-API-KEY - Request here https://nvd.nist.gov/developers/request-an-api-key
+
+        Returns:
+            List of NIST CPE URLs that may contain platform data.
+
+        Examples:
+            >>> OSPlatform('cisco','nxos','15.1(7)').get_nist_urls('YOURKEY')
+            ['https://services.nvd.nist.gov/rest/json/cpes/1.0?apiKey=YOURKEY&addOns=cves&cpeMatchString=cpe:2.3:o:cisco:nxos:15.1\\(7\\):*:*:*:*:*:*:*']
+        """
+        nist_urls = []
+        escape_list = [r"\(", r"\)"]
+        base_url = f"""https://services.nvd.nist.gov/rest/json/cpes/1.0?apiKey={api_key}&addOns=cves&cpeMatchString=cpe:2.3:o:"""
+
+        if self.platform in custom_nist_platforms.get(self.vendor.lower(), []):
+            return self.get_nist_urls(api_key)
+
+        for escape_char in escape_list:
+            if re.search(escape_char, self.version):
+                self.version = re.sub(escape_char, "\\" + escape_char, self.version)
+        nist_urls.append(f"{base_url}{self.vendor}:{self.platform}:{self.version.replace('-', ':')}:*:*:*:*:*:*:*")
+
+        return nist_urls
+
+
+class JuniperPlatform(OSPlatform):  # pylint: disable=R0902
```

Comment:
Wondering if we can use dataclasses instead.

Reply:
Dataclasses were put into place.

Resolving.